sys/arduino/base.cpp

```diff
@@ -58,3 +61,40 @@ void delay(unsigned long msec)
 {
     xtimer_usleep(1000 * msec);
 }
+
+#ifdef ADC_NUMOF
+int analogRead(int adc_pin)
+{
+    int adc_value;
+
+    /* State of ADC channel.
+     * 0: Not initialized
+     * 1: Successfully initialized
+     */
+    static int adc_state[ADC_NUMOF];
+
+    /* Check if the ADC line is valid */
+    if ((adc_pin < 0) || (adc_pin >= ADC_NUMOF)) {
+        return -1;
+    }
+
+    /* Initialization of given ADC channel */
+    if (adc_state[adc_pin] == 0) {
+        if (adc_init(ADC_LINE(adc_pin)) == 0) {
```

Comment:
As memo for a follow-up PR: we might as well generalize the `.../arduino-board.h` files to a central location and use `ifdefs` to create the actual pin mapping arrays...